modules/core/test/test_intrin_utils.hpp

```diff
@@ -1517,6 +1561,9 @@ void test_hal_intrin_uint64()
     TheTest<v_uint64>()
         .test_loadstore()
         .test_addsub()
+#if CV_SIMD_64F
+        .test_cmp64()
```

Comment:
Perhaps we should avoid this check for `v_uint64` types.
@terfendail Could you take a look?

Reply:
The problem is for NEON comparison of v_int64x2 and v_uint64x2 does NOT exist unless CV_SIMD_64.

Reply:
You are right.
In theory, `CV_SIMD_64F` should guard using of vectors of `doubles` and their operations.
Additionally, it guards some 64u/64s operations and some operations with 32f (like div) on NEON.

Perhaps we should address that in a separate PR.

Reply:
On the first look It seems like there is no reason to guard 64u/64s for NEON. This intrinsics doesn't use specific 64f related instructions. Maybe it make sense to change NEON guard scope?

Reply:
Underlying instructions are available on AARCH64 only. Removing guard breaks ARMv7 builds.

Probably we need dedicated macro for this functionality (btw, `==` can be emulated through 32-bit (like SSE2/3), but less/greater comparisons are not easy to re-implement).